pubsublite/consumer.go

```diff
@@ -32,6 +32,19 @@ import (
 	"github.com/elastic/apm-queue/queuecontext"
 )
 
+const (
+	// AtMostOnceDeliveryType acknowleges the message as soon as it's received
+	// and decoded, without waiting for the message to be processed.
+	AtMostOnceDeliveryType DeliveryType = iota
+	// AtLeastOnceDeliveryType acknowledges the message after it has been
+	// processed. It may or may not create duplicates, depending on how batches
+	// are processed by the underlying model.BatchProcessor.
+	AtLeastOnceDeliveryType
+)
```

Comment:
Should we have the `DeliveryType` for all queues?

Reply:
Yes, IIUC franz-go's default is to auto-commit the prior poll's records, on each poll. We could keep that for AtLeastOnce, and otherwise disable autocommit and call CommitUncommittedOffsets prior to calling Fetches.EachRecord.

Reply:
I can move the DeliveryType to `apmqueue` and remove the constructors that we have there since I don't think they're useful and we won't use them much. That way, we can reuse these in all the implementations.

Reply:
++ SGTM. If we want to reintroduce the constructor later, it could be in a separate package that imports pubsublite & kafka

Reply:
moved the DeliveryType to the root package.

Reply:
Opened https://github.com/elastic/apm-queue/issues/12 for kafka